pkg/sources/udsource/user_defined_source.go

```diff
@@ -194,6 +194,10 @@ func (u *userDefinedSource) PublishSourceWatermarks(msgs []*isb.ReadMessage) {
 	}
 }
 
+func (u *userDefinedSource) PublishIdleWatermarks(wm time.Time) {
+	// won't be used
+}
+
```

Comment:
iterate over the publishers and publish the idle watermark

Reply:
Publisher is not an array to iterate. I think it should be like below?
```
	publisher := u.loadSourceWatermarkPublisher(0)
	publisher.PublishWatermark(wmb.Watermark(t), nil, 0) // Source publisher does not care about the offset
```

Reply:
https://github.com/numaproj/numaflow/blob/112bc3b67ddc7433ac50988a16725853d91d32f2/pkg/sources/udsource/user_defined_source.go#L70
It is a map, we can iterate over the  map and publish the watermarks